tests/core/policies/test_intent_ted_policy.py

```diff
@@ -197,29 +267,99 @@ async def test_post_training_threshold_computation(
             training_trackers, default_domain, interpreter
         )
 
-        trained_policy.calculate_label_thresholds_post_training(
+        trained_policy.compute_label_quantiles_post_training(
             training_model_data, label_ids
         )
 
+        computed_thresholds = trained_policy.label_quantiles
+
         # -1 is used for padding and hence is not expected in the keys
         expected_keys = list(np.unique(label_ids))
         expected_keys.remove(-1)
 
-        assert sorted(list(trained_policy.label_thresholds.keys())) == sorted(
-            expected_keys
+        assert sorted(list(computed_thresholds.keys())) == sorted(expected_keys)
+
+    @pytest.mark.parametrize(
+        "tolerance, expected_thresholds",
+        [
+            (0.0, [0.2, -0.1]),
+            (0.30, [-0.1, -0.1]),
+            (0.75, [-2.9, -0.1]),
+            (0.72, [-2.7, -0.1]),
+            (0.78, [-2.9, -0.1]),
+            (1.0, [-4.1, -0.1]),
+        ],
+    )
+    def test_pick_thresholds_for_labels(
+        self, tolerance: float, expected_thresholds: List[float]
+    ):
+        label_id_tolerance_thresholds = {
+            0: [
+                0.2,
+                0.2,
+                0.2,
+                0.2,
+                0.2,
+                0.2,
+                -0.1,
+                -0.1,
+                -0.5,
+                -0.5,
+                -1.2,
+                -1.2,
+                -2.3,
+                -2.3,
+                -2.7,
+                -2.9,
+                -3.2,
+                -3.2,
+                -4.1,
+                -4.1,
+            ],
+            3: [
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+                -0.1,
+            ],
+        }
```

Comment:
This test case feels somewhat unnecessary since the value never changes. Can you make a sequence of monotonically decreasing floats distinct from the other one?

Reply:
> This test case feels somewhat unnecessary since the value never changes.

That's the purpose of the test to check if the threshold is picked correctly even if there is just one unique value for all tolerances for a particular label id.

> Can you make a sequence of monotonically decreasing floats distinct from the other one?

What would we be testing there? Do you want to see if the threshold is picked correctly when all thresholds corresponding to every possible value of tolerance is unique?

Reply:
Added a test where all quantiles are unique and monotonically decreasing.

Reply:
On second thought, I think what you meant was checking if the correct thresholds are picked for all possible values of tolerance, that is `range(0.0, 1.1, 0.05)`. Is that correct?

Reply:
No you understood me right the first time but I see your point now with this statement:

> the purpose of the test to check if the threshold is picked correctly even if there is just one unique value

so withdrawing my comment.

Reply:
But also appreciate the extra test :) !